apps/studio/components/interfaces/Functions/EdgeFunctionSecrets/AddNewSecretForm.tsx

```diff
@@ -0,0 +1,204 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useRef, useState } from 'react'
+import { SubmitHandler, useFieldArray, useForm } from 'react-hook-form'
+import { toast } from 'sonner'
+import z from 'zod'
+
+import { useParams } from 'common'
+import Panel from 'components/ui/Panel'
+import { useSecretsCreateMutation } from 'data/secrets/secrets-create-mutation'
+import { Eye, EyeOff, MinusCircle } from 'lucide-react'
+import {
+  Button,
+  Form_Shadcn_,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  FormItem_Shadcn_,
+  FormLabel_Shadcn_,
+} from 'ui'
+import { Input } from 'ui-patterns/DataInputs/Input'
+
+interface AddNewSecretFormProps {
+  onComplete?: () => void
+}
+
+type SecretPair = {
+  name: string
+  value: string
+}
+
+const AddNewSecretForm = ({ onComplete }: AddNewSecretFormProps) => {
+  const { ref: projectRef } = useParams()
+  const submitRef = useRef<HTMLButtonElement>(null)
+  const [showSecretValue, setShowSecretValue] = useState(false)
+
+  const FormSchema = z.object({
+    secrets: z.array(
+      z.object({
+        name: z
+          .string()
+          .min(1, 'Please provide a name for your secret')
+          .refine((value) => !value.match(/^(SUPABASE_).*/), {
+            message: 'Name must not start with the SUPABASE_ prefix',
```

Comment:
The error is rendering weird in FF.

![Screenshot 2025-02-06 at 15 08 08](https://github.com/user-attachments/assets/9042729d-c151-4468-90a8-2e91a7464410)